tsl/test/expected/dimension_range.out

```diff
@@ -0,0 +1,282 @@
+       		generate_series(1, 8, 1 ) AS g2(sensor_id)
+       	ORDER BY
+       		time;
+\set start_date '2023-03-17 17:51:11.322998+05:30'
+-- insert into new chunks
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 12, 21.98, 33.123, 'new row1');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 12, 17.66, 13.875, 'new row1');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 13, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 9, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 14, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 15, 0.988, 33.123, 'new row3');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 16, 4.6554, 47, 'new row3');
+-- Non-int, date, timestamp cannot be specified as a dimension ranges for now
+-- We could expand to FLOATs, NUMERICs later
+\set ON_ERROR_STOP 0
+SELECT * FROM add_dimension_range('sample_table', 'name');
+ERROR:  data type "character varying" unsupported for range calculation
+\set ON_ERROR_STOP 1
+-- Specify a dimension range
+SELECT * FROM add_dimension_range('sample_table', 'sensor_id');
+ dimension_range_id | created 
+--------------------+---------
+                  1 | t
+(1 row)
+
+-- The above should add an entry with MIN/MAX int64 entries for invalid chunk id
+-- to indicate that ranges on this column should be calculated for chunks
+SELECT * from _timescaledb_catalog.dimension_range;
+ id | hypertable_id | chunk_id | column_name | column_type |     range_start      |      range_end      | valid 
+----+---------------+----------+-------------+-------------+----------------------+---------------------+-------
+  1 |             1 |        0 | sensor_id   | integer     | -9223372036854775808 | 9223372036854775807 | t
+  2 |             1 |        1 | sensor_id   | integer     | -9223372036854775808 | 9223372036854775807 | t
+  3 |             1 |        2 | sensor_id   | integer     | -9223372036854775808 | 9223372036854775807 | t
+(3 rows)
+
+-- Skipping should work
+SELECT * FROM add_dimension_range('sample_table', 'sensor_id', true);
+NOTICE:  dimension range entry already exists for column "sensor_id", skipping
+ dimension_range_id | created 
+--------------------+---------
+                  1 | f
+(1 row)
+
+-- A query using a WHERE clause on "sensor_id" column will scan all the chunks
+:PREFIX SELECT * FROM sample_table WHERE sensor_id > 9;
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Append
+   ->  Index Scan using _hyper_1_1_chunk_sense_idx on _hyper_1_1_chunk
+         Index Cond: (sensor_id > 9)
+   ->  Index Scan using _hyper_1_2_chunk_sense_idx on _hyper_1_2_chunk
+         Index Cond: (sensor_id > 9)
+(5 rows)
+
+-- For the purposes of dimension ranges, a compressed chunk is considered as a
```

Comment:
Why don't we create a utility function that one can use to manually update the ranges? It would be useful for testing, but also for users that want to calculate ranges on any chunk, even if not compressed.

Refresh a particular column's range:
```
SELECT refresh_dimension_range(:chunk, 'sensor_id') from show_chunks('mytable') chunk;'
```

Refresh all columns on chunk:

```
SELECT refresh_dimension_range(:chunk) from show_chunks('mytable') chunk;'
```

Reply:
We could provide such a function. But am not sure about the utility of such a function on regular chunks. Regular chunks have DML in them and till we find a way to update these ranges efficiently the value add for them is pretty less.